src/viam/sdk/registry/registry.hpp

```diff
@@ -58,6 +58,37 @@ class ResourceRegistration {
     const google::protobuf::ServiceDescriptor* service_descriptor_;
 };
 
+// TODO(RSDK-3030): Potentially make ResourceRegistration2 the one and only
+// form of resource registration.
+template <typename ResourceClientT,
+          typename ResourceServerT,
+          typename ProtoServiceT,
+          typename ResourceRegT>
+class ResourceRegistration2 : public ResourceRegistration {
```

Comment:
I threw `ResourceRegistration2` into `registry.hpp` thinking that was a good enough "central place" for now. I think we can reconsider this class and its placement in RSDK-3030.

Reply:
To my way of looking at it RSDK-3030 is somewhat independent: it is about the practice of using file-scoped static initializers to make registrations happen, rather than the API of the Registry. But I'm willing to to stretch the scope a little. It is good that the TODO(RSDK-3030) is here, but can we please add a note or update the description of RSDK-3030 so that it includes revisiting this class? I suspect a large part of this can be boiled away into a template method on `Registry` itself, with the majority, if not all, of the details hidden inside `registry.cpp`.

Reply:
Good point; updated the description of https://viam.atlassian.net/browse/RSDK-3030.